src/dc_msg.rs

```diff
@@ -825,105 +825,74 @@ pub unsafe fn dc_msg_get_summary<'a>(
     )
+    .strdup()
 }
 
-/* the returned value must be free()'d */
-#[allow(non_snake_case)]
-pub unsafe fn dc_msg_get_summarytext_by_raw(
+/// Returns a summary test.
+pub fn dc_msg_get_summarytext_by_raw(
     type_0: Viewtype,
-    text: *const libc::c_char,
+    text: &Option<String>,
     param: &mut Params,
-    approx_characters: libc::c_int,
+    approx_characters: usize,
     context: &Context,
-) -> *mut libc::c_char {
-    /* get a summary text, result must be free()'d, never returns NULL. */
-    let mut ret;
-    let mut prefix: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut pathNfilename: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut value: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut append_text: libc::c_int = 1i32;
-    match type_0 {
-        Viewtype::Image => prefix = context.stock_str(StockMessage::Image).strdup(),
-        Viewtype::Gif => prefix = context.stock_str(StockMessage::Gif).strdup(),
-        Viewtype::Video => prefix = context.stock_str(StockMessage::Video).strdup(),
-        Viewtype::Voice => prefix = context.stock_str(StockMessage::VoiceMessage).strdup(),
+) -> String {
+    let mut append_text = true;
+    let prefix = match type_0 {
+        Viewtype::Image => context.stock_str(StockMessage::Image).into_owned(),
+        Viewtype::Gif => context.stock_str(StockMessage::Gif).into_owned(),
+        Viewtype::Video => context.stock_str(StockMessage::Video).into_owned(),
+        Viewtype::Voice => context.stock_str(StockMessage::VoiceMessage).into_owned(),
         Viewtype::Audio | Viewtype::File => {
             if param.get_int(Param::Cmd) == Some(6) {
-                prefix = context.stock_str(StockMessage::AcSetupMsgSubject).strdup();
-                append_text = 0i32
+                append_text = false;
+                context
+                    .stock_str(StockMessage::AcSetupMsgSubject)
+                    .to_string()
             } else {
-                pathNfilename = param
-                    .get(Param::File)
-                    .unwrap_or_else(|| "ErrFilename")
-                    .strdup();
-                value = dc_get_filename(pathNfilename);
-                let label = CString::new(
-                    context
-                        .stock_str(if type_0 == Viewtype::Audio {
-                            StockMessage::Audio
-                        } else {
-                            StockMessage::File
-                        })
-                        .as_ref(),
-                )
-                .unwrap();
-                prefix = dc_mprintf(
-                    b"%s \xe2\x80\x93 %s\x00" as *const u8 as *const libc::c_char,
-                    label.as_ptr(),
-                    value,
-                )
+                let value = unsafe {
+                    let base_fname = CString::yolo(param.get(Param::File).unwrap_or("ErrFilename"));
+                    to_string(dc_get_filename(base_fname.as_ptr()))
```

Comment:
you can use [`Path::filename`](https://doc.rust-lang.org/std/path/struct.Path.html#method.file_name) instead of the unsafe `dc_get_filename`